src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommandProviders.cs

```diff
@@ -118,12 +147,20 @@ public class RestoreCommandProviders
                 remoteProviders.Add(provider);
             }
 
+            var vulnerabilityInfoProviders = new List<IVulnerabilityInformationProvider>(sources.Count());
```

Comment:
Would it be possible to replace the usage of `sources.Count()` with `remoteProviders.Count` since the latter has a constant time complexity?